### ISSUE TITLE
(Android armeabi/arm64) Rune does not advance after choosing media directory

### ISSUE THREAD

**Issue by nesper8** [Losses/rune#214]
**Describe the bug**
Instead of moving to the next page Rune stays on the same page which promts the user to choose where your songs are located

**To Reproduce**
1. Install (affected on both) and run Rune
2. Click the 'Select Directory' button 
3. Assign media directory
4. See Rune staying on the same page

**Expected behavior**
Expected Rune to move to the next page 

**Screenshots**
https://github.com/user-attachments/assets/a70dbf2e-4819-4639-890b-bd6302f87252

**Smartphone:**
 - Device: Motorola Moto G32
 - OS: Android 13

**Comment by Losses** [comment]
As the file name and readme suggest, the Android version is not feature-complete and is still under active development. Thank you for your interest and support!

**Comment by SMNFXCN** [comment]
on my HiBy M300, I can proceed if I choose the Music folder on the internal storage. Choosing the Music folder on the microSD card does not let me proceed. It also doesn't add it as an extra library when I add a library within settings.

**Comment by Losses** [comment]
> on my HiBy M300, I can proceed if I choose the Music folder on the internal storage. Choosing the Music folder on the microSD card does not let me proceed. It also doesn't add it as an extra library when I add a library within settings.
> 
> edit: Other music players do not have issues accessing the microSD card. I've tested the latest stable and latest pre-release, and the latest pre-release does ask permission to access files, but the latest stable does not. Still isn't able to access the external storage.

Yes, this is a known issue. The reason is that the Rust part hasn't yet completed handling file system permissions. This is quite complex, and we are still working on it.

**Comment by nesper8** [comment]
Mind if I keep this open until you guys are ready?

**Comment by Losses** [comment]
Sure, let's keep it open!

**Comment by Losses** [comment]
@dheemanthm2003 As it's shown in the README, android builds are highly experimental. All fs related things are still buggy, we are finding solution to integrate Rust and Android correctly.

**Comment by Losses** [comment]
Hello everyone,

Our team has invested considerable time conducting in-depth research into the Rust and Android filesystem interoperability challenges. This is an exceptionally rare and difficult area to work with, but fortunately, we have recently completed the implementation of the underlying abstraction layer and are now working on integration.

Given that virtually every file I/O related function needs to be adapted and modified, this will be an extremely large-scale engineering effort. I'm hopeful that the issues raised in this thread will be properly addressed within the next few weeks.
Thank you once again for providing such valuable information and feedback.
